Entities/Block/Seat/Seat.as

```diff
@@ -461,7 +461,6 @@ void onTick(CBlob@ this)
 				{
 					CBlob@[] fireCannons;
 					Vec2f aim = occupier.getAimPos() - this.getPosition();//relative to seat
-					CBlob@ fitCannon = getMap().getBlobAtPosition(occupier.getAimPos());
 					
```

Comment:
No need to rollback!
This code is good, it is just missing a few checks...
Roll back this rollback and then add in code such as this ⬇️  at line 465
```
if (fitCannon !is null)
{
	//blob has to be on our ship and must be a cannon
	if (!fitCannon.hasTag("cannon") || fitCannon.getShape().getVars().customData != seatColor)
		@fitCannon = null;
}
```
If we don't check if the blob is a cannon, the seat will send a command to that blob regardless and it will error/warn you in console. If we don't check what ship ID the cannon is on, then you can fire other ship's cannons like a radio!